File: Resources/imobile_for_SP_app_iOS_SDK_2.0.24/sample/Objective-C/imobilesample/viewcontlloer/NativeViewController.m
//
//  NativeViewController.h
//  imobilesample
//
//  Created by i-mobile on 2015/08/28.
//  Copyright (c) 2015年 i-mobile. All rights reserved.
//

#import "NativeViewController.h"

#import <ImobileSdkAds/ImobileSdkAds.h>

@implementation NativeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSString *pid = self.adDisplayInfo.pid;
    NSString *mid = self.adDisplayInfo.mid;
    NSString *sid = self.adDisplayInfo.sid;
    
    // スポット情報を設定します
    [ImobileSdkAds registerWithPublisherID:pid MediaID:mid SpotID:sid];
    
    // デリゲートの登録をします
    [ImobileSdkAds setSpotDelegate:sid delegate:self];
    
    // 広告の取得を開始します
    [ImobileSdkAds startBySpotID:sid];

    // 広告を表示します
    [ImobileSdkAds getNativeAdData:sid ViewController:self Delegate:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - i-mobileデリゲードメソッド
- (void)onNativeAdDataReciveCompleted:(NSString *)spotId nativeArray:(NSArray *)nativeArray{
    
    // バナータイプへの表示
    ImobileSdkAdsNativeObject *ads = (ImobileSdkAdsNativeObject *)[nativeArray objectAtIndex:0];
    
    // タイトルの取得
    self.adTitleLabel.text = [ads getAdTitle];
    
    // ディスクリプションの取得
    self.adDescriptionLabel.text = [ads getAdDescription];
    
    //　画像の取得
    [ads getAdImageCompleteHandler:^(UIImage *loadimg) {
        //　画像の取得完了時に呼ばれます。
        self.adImageView.image = loadimg;
    }];
    
    // 該当するUIにクリックイベントの付加をします。
    [ads addClickFunction:self.adImageView];
    [ads addClickFunction:self.adTitleLabel];
    [ads addClickFunction:self.adDescriptionLabel];
}

@end

File: Resources/imobile_for_SP_app_iOS_SDK_2.0.24/sample/Swift/ImobileSample-Swift/class/NativeViewController.swift
//
//  NativeViewController.swift
//  ImobileSample-Swift
//
//  Created by i-mobile on 2015/11/26.
//  Copyright © 2016年 i-mobile. All rights reserved.
//
import UIKit

class NativeViewController: UIViewController,IMobileSdkAdsDelegate {
    
    var adDisplayInfo : AdDisplayInfo!
    
    // 広告の表示を行うUI
    @IBOutlet weak var adImageView: UIImageView!
    @IBOutlet weak var adTitleLabel: UILabel!
    @IBOutlet weak var adDescriptionLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let pid : String = adDisplayInfo.pid!
        let mid : String = adDisplayInfo.mid!
        let sid : String = adDisplayInfo.sid!
        
        // スポット情報を設定します
        ImobileSdkAds.registerWithPublisherID(pid, mediaID:mid, spotID:sid)
        // 広告の取得を開始します
        ImobileSdkAds.startBySpotID(sid)
        // 広告を表示します
        ImobileSdkAds.getNativeAdData(sid, viewController: self, delegate: self)
    }
    
    // ネイティブ広告の準備が完了した際に呼ばれます
    func onNativeAdDataReciveCompleted(spotId: String!, nativeArray: [AnyObject]!) {
        // 広告リストから広告の抽出
        let adObj = nativeArray[0] as! ImobileSdkAdsNativeObject
        
        // タイトルの設定
        adTitleLabel.text = adObj.getAdTitle()
        
        // ディスクリプションの設定
        adDescriptionLabel.text = adObj.getAdDescription()
        
        // 画像の読み込み
        adObj.getAdImageCompleteHandler({ (loadImg : UIImage!) -> Void in
            // 画像の読み込み完了時に実行されます
            
            // 画像の設定
            self.adImageView.image = loadImg
        })
        
        // クリックイベントの埋め込み
        adObj.addClickFunction(adTitleLabel)
        adObj.addClickFunction(adDescriptionLabel)
        adObj.addClickFunction(adImageView)
    }
}